electron/src/routes.ts

```diff
@@ -319,12 +319,19 @@ module.exports = {
     },
 
     patchSwap: (data: { local: boolean, swapID: string, swapContent?: string }, callback) => {
-        if (data.swapContent === null) {
-            swapController.remove(data.swapID, callback);
-            return;
-        }
+        repositoryLoad.then(() => {
+            let credentials;
+            if (repository.local.activeCredentials) {
+                credentials = repository.local.activeCredentials.id;
+            }
 
-        swapController.write(data.swapID, data.swapContent, callback);
+            if (data.swapContent === null) {
+                swapController.remove(credentials && !data.local ? (credentials + "/") : "" + data.swapID, callback);
```

Comment:
- Rename `credentials` to `credentialsID` or something similar. `credentials` is often used to extract `repository.credentials`. 
- If `data.local === false` and there is no active user, it would form a swap name for the local file. If it's a request for `data.local === false`, check for `activeCredentials` should throw an error if they are not set (`callback(new Error("..."))`).
- Complex and duplicate expression for determining the file name.
Extract it before using it. 
```const swapFileName = data.local ? data.swapID : `${credentialsID}/${data.swapID}` ```
- Addition has higher precedence than ternary conditional, credentials would not become a prefix of `data.swapID`
`false ? "foo" : "bam" + "boo" === "foo"// "bamboo"`
`true ? "foo" : "bam" + "boo" === "foo" // "foo", not "fooboo"`